plenum/server/batch_handlers/audit_batch_handler.py

```diff
@@ -98,7 +98,7 @@ def _create_audit_txn_data(self, three_pc_batch, last_audit_txn):
 
             # 3. ledger root (either root_hash or seq_no to last changed)
             # TODO: support setting for multiple ledgers
-            self.__fill_ledger_root_hash(txn, three_pc_batch, lid, last_audit_txn)
+            self.__fill_ledger_root_hash(txn, lid, ledger, last_audit_txn)
 
         # 4. state root hash
         txn[AUDIT_TXN_STATE_ROOT][three_pc_batch.ledger_id] = Ledger.hashToStr(three_pc_batch.state_root)
```

Comment:
We need to update `AUDIT_TXN_STATE_ROOT` for all changed ledgers as well.

Reply:
Fixed